pkg/kurl/join_cert.go

```diff
@@ -180,12 +205,10 @@ func getPodSpec(clientset kubernetes.Interface, namespace string) (*corev1.Pod,
 		},
 		Spec: corev1.PodSpec{
 			SecurityContext: &securityContext,
-			NodeSelector: map[string]string{
-				"node-role.kubernetes.io/master": "",
-			},
+			NodeSelector:    nodeSelector,
```

Comment:
lets change this to nodeAffinity and add both rather than detecting version

https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity

> If you specify multiple nodeSelectorTerms associated with nodeAffinity types, then the Pod can be scheduled onto a node if one of the specified nodeSelectorTerms can be satisfied.

```
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchExpressions:
          - key: node-role.kubernetes.io/control-plane
            operator: Exists
        - matchExpressions:
          - key: node-role.kubernetes.io/master
            operator: Exists
```

Reply:
Tested this